source/MaterialXCore/Node.cpp

```diff
@@ -252,14 +252,21 @@ vector<ElementPtr> NodeGraph::topologicalSort() const
 
     const vector<ElementPtr>& children = getChildren();
 
-    // Calculate in-degrees for all nodes, and enqueue those with degree 0.
-    std::unordered_map<ElementPtr, size_t> inDegree(children.size());
+    // Calculate in-degrees for all children.
+    std::unordered_map<ElementPtr, int> inDegree(children.size());
```

Comment:
Since this situation could only be caused by a logic error in the library (rather than a data error in a MaterialX document), my sense is that it should be handled by our unit tests, rather than being an Exception that NodeGraph::topologicalSort might throw.

What would you think about the following, which prevents the logic error case (however unlikely) from generating an underflow?

```
if (inDegree[downstreamElem] > 1)
{
    inDegree[downstreamElem]--;
}
else
{
    inDegree[downStreamElem] = 0;
    childQueue.push_back(downstreamElem);
}
```